lib/em-websocket/websocket.rb

```diff
@@ -42,8 +42,10 @@ def self.start(options, &blk)
 
     # Start WebSocket server inside eventmachine run loop
     def self.run(options)
-      host, port = options.values_at(:host, :port)
-      EM.start_server(host, port, Connection, options) do |c|
+      host, port, connection = options.values_at(:host, :port, :handler)
+      connection ||= Connection
```

Comment:
Is this really necessary for your example? As in, you could have just reopened Connection without passing it in. Practically speaking, you'll want to use provided Connection handler anyway, since that's where the bulk of the logic lives: https://github.com/igrigorik/em-websocket/blob/master/lib/em-websocket/connection.rb